src/viam/examples/project/pkg-config/Makefile

```diff
@@ -1,19 +1,24 @@
 CXX = c++
 
-CXXFLAGS = -std=c++14
-CXXFLAGS += `pkg-config --cflags viam-cpp-sdk-libviamsdk`
-LINKFLAGS = `pkg-config --libs viam-cpp-sdk-libviamsdk`
+VIAM_PREFIX ?= ../../../../../build/install
+PKG_CONFIG_PATH ?= $(VIAM_PREFIX)/lib/pkgconfig
```

Comment:
Could we instead just have a hard requirement that you run the makefile with `VIAM_PREFIX` set, and fail with an error if it isn't?

There is no requirement that the installation was made to any given location. Although `../../../../../build/install` is a guess that will be often correct, I think it ends up doing users a disservice to guess for them, when no such helpful guess can be provided in ordinary usage.

Or maybe use `pkg-config --exists viam-cpp-sdk-libviamsdk` to check if the Viam C++ SDK is actually in `PKG_CONFIG_PATH` and error out with a helpful message if not? That message could suggest the path `../../../../../build/install` or similar.

 It wouldn't be unreasonable, after all, for someone to build these examples against their package manager installed version of the C++ SDK! In that case, pkg-config would just find it.

Reply:
Second approach sounds good to me!

Reply:
So I just noticed that an invocation of, eg, `make example_dial` will return the following if `PKG_CONFIG_PATH` isn't set:

```
make example_dial
c++ -std=c++17 `pkg-config --cflags viam-cpp-sdk-libviamsdk` -c -o example_dial.o ../../dial/example_dial.cpp
Package viam-cpp-sdk-libviamsdk was not found in the pkg-config search path.
Perhaps you should add the directory containing `viam-cpp-sdk-libviamsdk.pc'
to the PKG_CONFIG_PATH environment variable
No package 'viam-cpp-sdk-libviamsdk' found
../../dial/example_dial.cpp:10:10: fatal error: 'boost/optional.hpp' file not found
#include <boost/optional.hpp>
         ^~~~~~~~~~~~~~~~~~~~
```

Maybe this is good enough? It looks like the alternative would be to write a `.PHONY` target to check `pkg-config --exists` and error out if not found, and then have all other targets depend on it...as usual it's a bit onerous to get Makefiles to express something fairly simple